### PR TITLE
feat: Hash table implementation 

### PR DESCRIPTION
**Pull request by apoorvcodes**

### CODE REVIEW

**Comment by apoorvcodes** [comment]
Thanks to our man @jfontan for this, he is real guy to be credited.

File: minima.go [hunk 1]
```diff
@@ -212,20 +220,20 @@ func (m *minima) Mount(path string, router *Router) *minima {
...
-func (m *minima) UseConfig(config *Config) *minima {
-	for _, v := range config.Middleware {
-		m.Middleware.plugin = append(m.Middleware.plugin, &Middleware{handler: v})
-	}
-	for _, rt := range config.Router {
-		m.router.UseRouter(rt)
-	}
-	return m
-}
+// /**
```
Comment (megatank58) [inline minima.go:223]: why is this commented? can't it just be removed if it's not needed

**Review by savioxavier — APPROVED** [review]

File: mux.go [hunk 2]
```diff
@@ -1,38 +1,175 @@
...
 }
 
 /**
-@info Match a path
-@param {string} [path] The path to match
-@returns {bool, map[]string[]string}
+@info The Route structure
+@property {string} [prefix] The prefix of the route
+@property {[]param} [partnames] The route paths split into parts
+@property {function} [Handler] The handler to be used
```
Comment (megatank58) [inline mux.go:21]:
```suggestion
@property {Handler} [function] The handler to be used
```

File: mux.go [hunk 3]
```diff
@@ -1,38 +1,175 @@
...
+	return &Routes{
+		roots: make(map[string][]Route),
+	}
+}
+
+/**
+@info Adds a new route to the routes table
+@param {string} [path] Path of the route
+@param {Handler} [handler] Handler of the route
+@returns {}
```
Comment (megatank58) [inline mux.go:51]: Just remove this if it returns nothing

File: router.go [hunk 4]
```diff
@@ -17,7 +16,8 @@ type Handler func(res *Response, req *Request)
 @property {map[string][]*mux} [route] The mux routes
 */
 type Router struct {
-	routes map[string][]*mux
+	notfound Handler
+	routes   map[string]*Routes
```
Comment (megatank58) [inline router.go:20]: docs need to be written for these

**Review by megatank58 — CHANGES_REQUESTED** [review]
Just some things, rest seems fine

**Review by megatank58 — APPROVED** [review]

**Review by savioxavier — APPROVED** [review]

**Review by apoorvcodes — COMMENTED** [review]
Fixed everything

**Pull request merged**